packages/mui-material/src/Rating/Rating.js

```diff
@@ -242,7 +242,8 @@ function RatingItem(props) {
   const isFocused = itemValue <= focus;
   const isChecked = itemValue === ratingValueRounded;
 
-  const id = useId();
+  const id = (name ? `${name}-` : '') + useId();
```

Comment:
used this instead of because eslint was complaining
```tsx
// error  React Hook "useId" is called conditionally. 
// React Hooks must be called in the exact same order in every component render  react-hooks/rules-of-hooks
const id = name ? `${name}-${useId()}` : useId();
```